client/txn.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// Author: Andrei Matei (andrei@cockroachlabs.com)
+
+// Package client provides helpers for using CockroachDB in client
+// applications.
+package client
+
+import (
+	"database/sql"
+
+	"github.com/cockroachdb/pq"
+)
+
+// ExecuteTxn runs fn inside a transaction and retries it as needed.
+// On non-retryable failures, the transaction is aborted and rolled
+// back; on success, the transaction is committed.
+//
+// For more information about CockroachDB's transaction model see
+// https://cockroachlabs.com/docs/transactions.html.
+//
+// NOTE: the supplied exec closure should not have external side
+// effects beyond changes to the database.
+func ExecuteTxn(db *sql.DB, fn func(*sql.Tx) error) error {
```

Comment:
We use `Txn` everywhere in cockroach-land, but the standard library uses `Tx`. Sort of awkward to see `Txn` and `Tx` in the same function definition.